compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5420,6 +5429,20 @@ public void visit(BLangBinaryExpr binaryExpr, AnalyzerData data) {
         data.resultType = types.checkType(binaryExpr, actualType, data.expType);
     }
 
+    private BType getXmlBinaryOpResultType(BType opType, BType constituentType, SymbolEnv env, Location pos) {
```

Comment:
This is meant to be called only when the other operand is of a type that is a subtype of string, right? Shall we make sure the method name reflects what happens here?